library/src/scala/tasty/reflect/CompilerInterface.scala

```diff
@@ -1024,8 +1026,20 @@ trait CompilerInterface {
 
   def isInstanceOfRecursiveType(given ctx: Context): IsInstanceOf[RecursiveType]
 
+  /** Create a RecType, normalizing its contents. This means:
+   *
+   *   1. Nested Rec types on the type's spine are merged with the outer one.
+   *   2. Any refinement of the form `type T = z.T` on the spine of the type
+   *      where `z` refers to the created rec-type is replaced by
+   *      `type T`. This avoids infinite recursions later when we
+   *      try to follow these references.
+   */
+  def RecursiveType_apply(parentExp: RecursiveType => Type)(given ctx: Context): RecursiveType
```

Comment:
I'm wondering if there are macros that use recursive types? Scala doesn't have syntax for recursive types.

Reply:
We will still get some types that are represented with recursive types in TASTy, hence we need to support it.